### ISSUE TITLE
Issue: AST-match should catch when Javadoc presence differs between example files*

### ISSUE THREAD

**Issue by smita1078** [checkstyle/checkstyle#20705]
When writing multiple `Example*.java` files for the same check, each example needs to match the same AST structure — the only allowed difference is the trailing `// violation` , // ok , // filtered violation comment.

The problem: if one example has `/** Javadoc. */` above a member and another doesn't , the diff/AST-match test should treat this as a structural mismatch.

Example that broke it:
```java
/** Javadoc. */
public interface F {}
/** Javadoc. */
public interface G {}
```
vs. a sibling example without those Javadoc blocks.

```java
/** Javadoc. */
public interface F {}

public interface G {}
```
TODO: Enhance AST Logic to validate javadoc mismatch as well

**Comment by smita1078** [comment]
on it

**Comment by romani** [comment]
Fix is merged
